project_issue_task_sync/README.rst

```diff
@@ -0,0 +1,35 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=====================
+Sync Issues and Tasks
+=====================
+
+This module syncs assigned user and state from task to issue and viceversa.
+If a task is created and does not have a issue a related issue will be
+generated. Projects may or may not have this syncing function, the activation
+of sync between issues and task may be activated on the project form by
+selecting the checkbox "sync issues and tasks". By default projects that use
+tasks and issues will be synced.
```

Comment:
>  issue and tasks are quite redundant logically, it seems to me like these to models are the result of an accidental duplication of effort

I see your pain, and this is worth discussing.
. I've also been having this debate for years now: should Issues and Tasks just be the same Model?
I first started with the Open Issue -> Create Task workflow.
It looked complex to users, so on a second phase I dropped Issues and used only Tasks, and just added triaging as Stages.
Now it seems that it is an oversimplifications and I'm feeling the need to go back to the first form: Issues -> Tasks.

I see Issues and customer/frontend facing, and Tasks as delivery/backend facing.
Issues are for Customers and 1st level Service Desk interaction.
Tasks are for 1st level Service Desk interaction and the actual people/teams performing the necessary work.
One Issue may need many Tasks to be preformed, possibly by different people.
For example, a "Create New User"  Issue would mean creating Tasks for the Sysadmin to create accounts, for HR to create access card, for IT to provide PC and phone, etc.

Keeping Issues a separate module makes it easier to manage access control, and keep internal messages on Tasks away from the sight of the end Customer.

I'm still thinking if the automatic Task creation improves the process. After you always need to assign the work to someone, and that is about the same effort a the "create Task" step.
And I'm not sure this behaves properly when you have multiple Tasks for the same Issue.
Perhaps having a "root Task" with "subTasks" can answer that?

What do you think?